host/xtest/regression_1000.c

```diff
@@ -902,6 +904,140 @@ static void xtest_tee_test_1011(ADBG_Case_t *c)
+	TEEC_Session session;
+	TEEC_Context context;
+};
+
+static TEEC_Result do_open_session(struct xtest_session *cs,
+				   const TEEC_UUID *uuid)
+{
+	TEEC_Result res = TEEC_ERROR_GENERIC;
+	uint32_t ret_orig = 0;
+
+	TEEC_InitializeContext(NULL, &cs->context);
+
+	res = TEEC_OpenSession(&cs->context,
+			       &cs->session, uuid,
+			       TEEC_LOGIN_PUBLIC, NULL, NULL,
+			       &ret_orig);
+	if (res != TEEC_SUCCESS) {
+		TEEC_FinalizeContext(&cs->context);
+		return res;
+	}
+
+	return res;
+}
+
+static TEEC_Result do_get_counter(struct xtest_session *cs,
+				  uint32_t *counter)
+{
+	TEEC_Result res = TEEC_ERROR_GENERIC;
+	TEEC_Operation op = TEEC_OPERATION_INITIALIZER;
+	uint32_t ret_orig = 0;
+
+	op.paramTypes = TEEC_PARAM_TYPES(TEEC_VALUE_OUTPUT,
+					 TEEC_NONE,
+					 TEEC_NONE,
+					 TEEC_NONE);
+
+	res = TEEC_InvokeCommand(&cs->session,
+				 TA_SIMS_CMD_GET_COUNTER, &op,
+				 &ret_orig);
+	*counter = op.params[0].value.a;
+
+	return res;
+}
+
+static TEEC_Result do_panic(struct xtest_session *cs)
+{
+	return TEEC_InvokeCommand(&cs->session, TA_SIMS_CMD_PANIC,
+				  NULL, NULL);
+}
+
+static void do_close_session(struct xtest_session *cs)
+{
+	TEEC_CloseSession(&cs->session);
+	TEEC_FinalizeContext(&cs->context);
+}
+
+static int test_ta_panic_behavior(ADBG_Case_t *c, const TEEC_UUID *uuid,
+				  bool multi_instance)
+{
+	TEEC_Result exp_res = TEEC_ERROR_GENERIC;
+	uint32_t counter = 0;
+	uint32_t exp_counter = 0;
+	uint32_t ret = 0;
+	struct xtest_session cs[3];
+
+	memset(cs, 0, sizeof(cs));
+
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c, do_open_session(&cs[0], uuid)))
+		goto bail;
+
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c, do_open_session(&cs[1], uuid)))
+		goto bail0;
+
+	if (!ADBG_EXPECT_TEEC_RESULT(c, TEEC_SUCCESS,
+				     do_get_counter(&cs[0], &counter)))
+		goto bail1;
+
+	if (!ADBG_EXPECT(c, 0, counter))
+		goto bail1;
+
+	if (!ADBG_EXPECT_TEEC_RESULT(c, 0,
+				     do_get_counter(&cs[1], &counter)))
+		goto bail1;
+
+	exp_counter = multi_instance ? 0 : 1;
+	if (ADBG_EXPECT(c, exp_counter, counter))
+		goto bail1;
+
+	if (!ADBG_EXPECT_TEEC_RESULT(c, TEE_ERROR_TARGET_DEAD,
+				     do_panic(&cs[1])))
+		goto bail1;
+
+	exp_res = multi_instance ? TEEC_SUCCESS : TEE_ERROR_TARGET_DEAD;
+	if (!ADBG_EXPECT_TEEC_RESULT(c, exp_res,
+				     do_get_counter(&cs[0], &counter)))
+		goto bail1;
+
+	if (!ADBG_EXPECT_TEEC_RESULT(c, TEE_ERROR_TARGET_DEAD,
+				     do_get_counter(&cs[1], &counter)))
+		goto bail1;
+
+	/* Open a new session on a new context */
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c, do_open_session(&cs[2], uuid)))
+		goto bail1;
+
+	/* Optional test, already tested above */
+	if (multi_instance) {
+		if (!ADBG_EXPECT_TEEC_RESULT(c, TEEC_SUCCESS,
+					     do_get_counter(&cs[0], &counter)))
+			goto bail2;
+
+		if (!ADBG_EXPECT(c, 0, counter))
+			goto bail2;
+	}
+
+	if (!ADBG_EXPECT_TEEC_RESULT(c, TEEC_SUCCESS,
+				     do_get_counter(&cs[2], &counter)))
+		goto bail2;
+
+	if (!ADBG_EXPECT(c, 0, counter))
+		goto bail2;
+
```

Comment:
add here `ret = 0;`

Reply:
You're right. this return code is useless. You can remove `ret` and make the function returning `void`.

Reply:
Done.